docs/README.iOS.md

```diff
@@ -140,27 +139,38 @@ make -j$(getconf _NPROCESSORS_ONLN) -C tools/depends/target/binary-addons ADDONS
 **[back to top](#table-of-contents)**
 
 ## 6. Build Kodi
+
+## 6.1. Generate Project Files
+
 Before you can use Xcode to build Kodi, the Xcode project has to be generated with CMake. CMake is built as part of the dependencies and doesn't have to be installed separately. A toolchain file is also generated and is used to configure CMake.
 
-### 6.1. Build with Xcode
-Create an out-of-source build directory:
+Change to Kodi's source code directory:
```

Comment:
do not remove the info, for how to do a out of source build

Reply:
what is preferred by team kodi. a true out of tree build, or cmakesys doing an "out of tree" build in a dir inside source?

i ask only because is it worthwhile adding the following to cmakebuildsys to allow build dir to be provided by user?
```
ifeq ($(BUILD_DIR),)
  BUILD_DIR=$(CMAKE_SOURCE_DIR)/build
endif
```

Reply:
For both are out of source builds, so everything which doesn't hit the following code.
https://github.com/xbmc/xbmc/blob/40da2b6ef0dbee65ddf90cdb375b1ad5bf5e772b/CMakeLists.txt#L20-L22

Your suggested code wouldn't be limited to buildir inside the source tree, so it's okay for me.
If you want to add the BUILD_DIR change, README.iOS.md, README.macOS.md and README.tvOS.md have to be adjusted. Additionally to the Xcode auto detection for iOS add and option to switch macOS to Xcode generator. 
Would be great if you add the info about the optional BUILD_DIR variable in README.Android.md and README.RaspberryPi.md.

Reply:
Ok, ill try tackle all that over the weekend.

Reply:
have added a 3rd commit for build_dir change. Have updated other docs as suggested.
Let me know what you think needs altering.